test/LEAF_Request_Portal_Tests/tests/api/FormEditorControllerTest.php

```diff
@@ -649,6 +649,87 @@ public function testSetCategoryPrivileges_addPriv() : void
+      $this->assertNotNull($category);
+      $this->assertEquals('form_f4687', $category['categoryID']);
+      $this->assertEquals('A Simple Sample Form', $category['categoryDescription']);
+
+      $category = self::$client->get('?a=formStack/categoryList/all')[2];
+      $this->assertNotNull($category);
+      $this->assertEquals('form_f4689', $category['categoryID']);
+      $this->assertEquals('A Staple form', $category['categoryDescription']);
+
+      self::$client->postEncodedForm('?a=formEditor/_form_f4687/stapled', array(
+          'stapledCategoryID' => $category['categoryID'],
+      ));
+
+      $category = self::$client->get('?a=formEditor/_form_f4687/stapled')[0];
+      $this->assertNotNull($category);
+      $this->assertEquals('form_f4689', $category['categoryID']);
+    }
+
+
+    /**
+     * Tests the GET `formEditor/[text]/stapled` endpoint.
+     */
+    public function testGetStapledCategories() : void
+    {
+      $category = self::$client->get('?a=formStack/categoryList/all')[1];
+      $this->assertNotNull($category);
+      $this->assertEquals('form_f4687', $category['categoryID']);
+      $this->assertEquals('A Simple Sample Form', $category['categoryDescription']);
+
+      $category = self::$client->get('?a=formStack/categoryList/all')[2];
+      $this->assertNotNull($category);
+      $this->assertEquals('form_f4689', $category['categoryID']);
+      $this->assertEquals('A Staple form', $category['categoryDescription']);
+
+      self::$client->postEncodedForm('?a=formEditor/_form_f4687/stapled', array(
+          'stapledCategoryID' => $category['categoryID'],
+      ));
+
+      $category = self::$client->get('?a=formEditor/_form_f4687/stapled')[0];
+      $this->assertNotNull($category);
+      $this->assertEquals('form_f4689', $category['categoryID']);
+    }
+
+    /**
+     * Tests the DELETE `formEditor/[text]/stapled/[text]` endpoint.
+     */
+    public function testRemoveStapledCategory() : void
+    {
+      $category = self::$client->get('?a=formStack/categoryList/all')[1];
+      $this->assertNotNull($category);
+      $this->assertEquals('form_f4687', $category['categoryID']);
+      $this->assertEquals('A Simple Sample Form', $category['categoryDescription']);
+
+      $category = self::$client->get('?a=formStack/categoryList/all')[2];
+      $this->assertNotNull($category);
+      $this->assertEquals('form_f4689', $category['categoryID']);
+      $this->assertEquals('A Staple form', $category['categoryDescription']);
+
+      self::$client->postEncodedForm('?a=formEditor/_form_f4687/stapled', array(
+          'stapledCategoryID' => $category['categoryID'],
+      ));
+
+      $category = self::$client->get('?a=formEditor/_form_f4687/stapled')[0];
+      $this->assertNotNull($category);
+      $this->assertEquals('form_f4689', $category['categoryID']);
+
+
+      $delResponse = self::$client->delete('?a=formEditor/_form_f4687/stapled/_form_f4689/');
```

Comment:
I can't get this test to pass either. Keep getting the error: "Invalid request. Missing Referer"

It seems to be actually deleting the stapled form, but I think the Guzzle method is completing correctly.

Reply:
you will have to disable "$this->verifyAdminReferrer();" on DELETE in LEAF_Request_Portal/api/controllers/FormEditorController.php to get this test passed. this an existing issue with GUZZLE.

Reply:
Got it.